### ISSUE TITLE
`safe_mint` and `safe_transfer` should require `_invokeNFTReceiver` to return `true`

### ISSUE THREAD

**Issue by marc0olo** [appinesshq/ae-nft#8]
I guess you would want this entrypoint to throw and error and fail if the recipient / receiver doesn't implement this interface or returns false.

see:
- https://github.com/appinesshq/ae-nft/blob/05e4ef914f2e8387eea59b6bc2db8f20f9b26b42/contracts/NFT.aes#L165
- https://github.com/appinesshq/ae-nft/blob/05e4ef914f2e8387eea59b6bc2db8f20f9b26b42/contracts/NFT.aes#L168

**Comment by marc0olo** [comment]
basically I mean the `require` is missing somewhere here. either in this method or where the method is invoked, e.g.:
- https://github.com/appinesshq/ae-nft/blob/05e4ef914f2e8387eea59b6bc2db8f20f9b26b42/contracts/NFT.aes#L115

**Comment by arjanvaneersel** [comment]
What I tried to say on Discord is that ERC-721 doesn't throw in case the receiver doesn't implement the method. So therefore I use the bool to indicate whether the call was successful or not. However what I do notice is that with the changes today I removed prolongating the bool result up. So instead of require and throw, it should at least return the bool, so that the developer of a contract can decide whether to throw or not.

**Comment by marc0olo** [comment]
let me rephrase the issue to be more clear. IMO an entrypoint called `safe_mint` or `safe_transfer` should require that check to be true. do you agree? this is what I tried to explain

**Comment by arjanvaneersel** [comment]
I agree with you, but it would mean that we break ERC-721 compatibility.

**Comment by marc0olo** [comment]
> I agree with you, but it would mean that we break ERC-721 compatibility.

can you elaborate on that one? I don't understand 😅 let's get first the discussion about the interface getting started in the AEX proposal and then digging deeper into implementation details

**Comment by arjanvaneersel** [comment]
You are right. It should throw if the call fails. I included this in the [AEX proposal](https://github.com/aeternity/AEXs/issues/141)
